spec/acceptance/upmark_spec.rb

```diff
@@ -341,4 +341,36 @@ def actual
       expect(html).to convert_to("Hi\nthere")
     end
   end
-end
+
+  context "nested unordered lists" do
+  let(:html) do
+      <<-HTML
+      <ul>
+        <ul>
+          <li>List item</li>
+        </ul>
+      </ul>
+      HTML
+    end
+
+    it "generates readable output" do
+      expect(html).to convert_to("* * List item")
+    end
+  end
+
+  context "nested ordered lists" do
+    let(:html) do
+        <<-HTML
+        <ol>
+          <ol>
+            <li>List item</li>
+          </ol>
+        </ol>
+        HTML
+      end
+
+      it "generates readable output" do
+        expect(html).to convert_to("1. 1. List item")
+      end
+    end
+  end
```

Comment:
Checking out these new tests to the `main` branch will raise the following errors in the test suite, as expected:

```shell
╰─$ bundle exec rspec                                              

Randomized with seed 12685
..F................F..........................................................................

Failures:

  1) Upmark.convert nested unordered lists generates readable output
     Failure/Error: children = element[:children].map {|value| value.strip != "" ? value : nil }.compact
     
     NoMethodError:
       undefined method `strip' for ["* List item\n"]:Array
     # ./lib/upmark/transform/markdown.rb:51:in `block (2 levels) in <class:Markdown>'
     # ./lib/upmark/transform/markdown.rb:51:in `map'
     # ./lib/upmark/transform/markdown.rb:51:in `block in <class:Markdown>'
     # ./lib/upmark/transform_helpers.rb:22:in `block (2 levels) in element'
     # ./lib/upmark.rb:20:in `convert'
     # ./spec/acceptance/upmark_spec.rb:8:in `actual'
     # ./spec/acceptance/upmark_spec.rb:4:in `block (3 levels) in <top (required)>'
     # ./spec/acceptance/upmark_spec.rb:357:in `block (3 levels) in <top (required)>'

  2) Upmark.convert nested ordered lists generates readable output
     Failure/Error: children = element[:children].map {|value| value.strip != "" ? value : nil }.compact
     
     NoMethodError:
       undefined method `strip' for ["1. List item\n"]:Array
     # ./lib/upmark/transform/markdown.rb:56:in `block (2 levels) in <class:Markdown>'
     # ./lib/upmark/transform/markdown.rb:56:in `map'
     # ./lib/upmark/transform/markdown.rb:56:in `block in <class:Markdown>'
     # ./lib/upmark/transform_helpers.rb:22:in `block (2 levels) in element'
     # ./lib/upmark.rb:20:in `convert'
     # ./spec/acceptance/upmark_spec.rb:8:in `actual'
     # ./spec/acceptance/upmark_spec.rb:4:in `block (3 levels) in <top (required)>'
     # ./spec/acceptance/upmark_spec.rb:373:in `block (3 levels) in <top (required)>'

Finished in 0.08328 seconds (files took 0.08653 seconds to load)
94 examples, 2 failures
```